pkg/clients/compute/aks.go

```diff
@@ -280,38 +129,40 @@ func newManagedCluster(c *v1alpha3.AKSCluster, appID, secret string) containerse
 				{
 					Name:   to.StringPtr(AgentPoolProfileName),
 					Count:  &nodeCount,
-					VMSize: containerservice.VMSizeTypes(c.Spec.NodeVMSize),
+					VMSize: to.StringPtr(c.Spec.NodeVMSize),
+					Mode:   containerservice.AgentPoolModeSystem,
```

Comment:
Even though system mode is set in here, I got the following error when I tried to create an AKS cluster with a user-managed identity 
```
Warning  CannotCreateExternalResource  96s (x25 over 3m21s)      managed/akscluster.compute.azure.crossplane.io  cannot create AKSCluster: containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="MustDefineAtLeastOneSystemPool" Message="Must define at least one system pool."
```
I used following manifest: 
```
apiVersion: compute.azure.crossplane.io/v1alpha3
kind: AKSCluster
metadata:
  name: ezgi-akscluster
  labels:
    example: "true"
spec:
  resourceGroupName: ezgi
  vnetSubnetIDRef:
    name: ezgi-sub
  location: West US 2
  version: "1.21.7"
  nodeCount: 1
  nodeVMSize: Standard_B2s
  dnsNamePrefix: ezgi-aks
  disableRBAC: false
  identity:
    identityNames:
    - ezgiidentity
    type: UserAssigned
  providerConfigRef:
    name: example
  writeConnectionSecretToRef:
    namespace: crossplane-system
    name: ezgi-akscluster
```

Reply:
Hi @ezgidemirel,
Thank you for giving this a try. It looks like to be an issue specific to provisioning AKS clusters as you have mentioned below that provisioning some other resources were successful. Using SP credentials, I tried provisioning an AKS cluster with success using the example manifest from this PR. You are using the provider package `ulucinar/provider-azure:build-3eacbce2`, right? 
I will also give it a try using a user-assigned managed identity in ProviderConfig and report back to you, although it should not be related.

Reply:
Hello again,
Using the provider package `ulucinar/provider-azure:build-3eacbce2`, on an AKS cluster with a user-assigned managed kubelet identity, I tried to provision an AKS cluster using the following manifests successfully:
```yaml
apiVersion: azure.crossplane.io/v1beta1
kind: ProviderConfig
metadata:
  name: example
spec:
  clientID: <client-id of the user-assigned managed kubelet identity>
  credentials:
    source: InjectedIdentity
  subscriptionID: <Azure subscription ID>

---

apiVersion: compute.azure.crossplane.io/v1alpha3
kind: AKSCluster
metadata:
  name: example-akscluster-alper2
  labels:
    example: "true"
spec:
  resourceGroupName: alper
  location: West US 2
  version: "1.21.7"
  nodeCount: 1
  nodeVMSize: Standard_B2s
  dnsNamePrefix: crossplane-aks
  disableRBAC: false
  identity:
    type: SystemAssigned
  providerConfigRef:
    name: example
  writeConnectionSecretToRef:
    namespace: crossplane-system
    name: example-akscluster
```

I have also verified accessing the provisioned AKS cluster using the credentials (kubeconfig) stored in the connection secret.

Reply:
The AKS cluster I tried to provision has `UserAssigned` managed identity. I believe that's the reason why my attempt is failing.

Reply:
Hi @ezgidemirel,
The following also succeeded in my cluster:
```yaml
apiVersion: compute.azure.crossplane.io/v1alpha3
kind: AKSCluster
metadata:
  name: example-akscluster-alper2
  labels:
    example: "true"
spec:
  resourceGroupName: alper
  location: West US 2
  version: "1.21.7"
  nodeCount: 1
  nodeVMSize: Standard_B2s
  dnsNamePrefix: crossplane-aks
  disableRBAC: false
  identity:
    type: UserAssigned
    identityNames:
    - control-plane-identity-alper
  providerConfigRef:
    name: example
  writeConnectionSecretToRef:
    namespace: crossplane-system
    name: example-akscluster
```

Reply:
Updating resource manifests to use the same location didn't work. However, removing `vnetSubnetIDRef` from the spec did the trick. I could successfully provision an AKS cluster.

Reply:
It turns out that we are not setting the default node pool's mode to System iff a subnet is specified :)
Fixed the bug and successfully provisioned an AKS cluster by specifying a subnet for its node pool.

Reply:
Thank you very much @ezgidemirel for finding this issue and helping me in fixing it. Very much appreciated!